server/src/main/java/org/opensearch/snapshots/SnapshotsService.java

```diff
@@ -2168,16 +2173,41 @@ private void deleteSnapshotsFromRepository(
             assert currentlyFinalizing.contains(deleteEntry.repository());
             final List<SnapshotId> snapshotIds = deleteEntry.getSnapshots();
             assert deleteEntry.state() == SnapshotDeletionsInProgress.State.STARTED : "incorrect state for entry [" + deleteEntry + "]";
-            repositoriesService.repository(deleteEntry.repository())
-                .deleteSnapshots(
-                    snapshotIds,
-                    repositoryData.getGenId(),
-                    minCompatibleVersion(minNodeVersion, repositoryData, snapshotIds),
-                    ActionListener.wrap(updatedRepoData -> {
-                        logger.info("snapshots {} deleted", snapshotIds);
-                        removeSnapshotDeletionFromClusterState(deleteEntry, null, updatedRepoData);
-                    }, ex -> removeSnapshotDeletionFromClusterState(deleteEntry, ex, repositoryData))
+            StepListener cleanupRemoteStoreLockFilesStep = new StepListener<>();
+            boolean cleanupRemoteStoreLockFiles = repositoriesService.repository(deleteEntry.repository()) instanceof BlobStoreRepository;
```

Comment:
nit: add comment here explaining why are we doing this only for BlobStoreRepository.

Reply:
Outdated change